pydantic/_internal/_generics.py

```diff
@@ -338,6 +336,34 @@ def replace_types(type_: Any, type_map: Mapping[Any, Any] | None) -> Any:
     return type_map.get(type_, type_)
 
 
+def has_instance_in_type(type_: Any, isinstance_target: Any) -> bool:
```

Comment:
Although this doesn't really relate to pydantic generics functionality, I put this function in `_generics.py` because it is very closely related in implementation to `_generics.replace_types`, and because the interesting parts of the logic are mostly related to "walking" generic types.